rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,209 @@
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            Coin maxTransferValueCoin = Coin.valueOf(maxTransferValue.longValueExact());
+            LockWhitelistEntry whiteListEntry = new OneOffWhiteListEntry(address, maxTransferValueCoin);
+
+            return this.addLockWhitelistAddress(tx, whiteListEntry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    @Override
+    public int addUnlimitedLockWhitelistAddress(Transaction tx, String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            LockWhitelistEntry whiteListEntry = new UnlimitedWhiteListEntry(address);
+
+            return this.addLockWhitelistAddress(tx, whiteListEntry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    /**
+     * Removes the given address from the lock whitelist.
+     *
+     * @param addressBase58 the base58-encoded address to remove from the whitelist
+     * @return 1 SUCCESS, -1 ADDRESS_NOT_EXIST, -2 INVALID_ADDRESS_FORMAT, -10 GENERIC_ERROR,
+     * 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int removeLockWhitelistAddress(Transaction tx, String addressBase58) {
+        if (isAuthorizedLockWhitelistChange(tx)) {
+            return GENERIC_ERROR.getCode();
+        }
+
+        LockWhitelist whitelist = whitelistStorageProvider.getLockWhitelist();
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+
+            if (!whitelist.remove(address)) {
+                return ADDRESS_NOT_EXIST.getCode();
+            }
+            return SUCCESS.getCode();
+        } catch (AddressFormatException e) {
+            return INVALID_ADDRESS_FORMAT.getCode();
+        } catch (Exception e) {
+            logger.error("Unexpected error in removeLockWhitelistAddress: {}", e.getMessage());
+            panicProcessor.panic("lock-whitelist", e.getMessage());
+            return UNKNOWN_ERROR.getCode();
+        }
+    }
+
+    /**
+     * Sets a delay in the BTC best chain to disable lock whitelist
+     *
+     * @param tx                  current RSK transaction
+     * @param disableBlockDelayBI block since current BTC best chain height to disable lock
+     *                            whitelist
+     * @return 1 SUCCESS, -1 DELAY_ALREADY_SET, -2 DISABLE_BLOCK_DELAY_INVALID, -10 GENERIC_ERROR
+     */
+    @Override
+    public int setLockWhitelistDisableBlockDelay(Transaction tx, BigInteger disableBlockDelayBI, int btcBlockchainBestChainHeight)
+        throws IOException, BlockStoreException {
```

Comment:
Where are these exceptions thrown?

Reply:
This is the trace of how it is handling the exception, nevertheless, on the Bridge, it is not clear how this is handled. It seems, it is not being handled properly:
`Bridge -> BridgeSupport -> WhitelistSupport`

Reply:
My concern is that I don't see any call inside `setLockWhitelistDisableBlockDelay` that is throwing any of those exceptions. Is there any or can they be removed from the method signature?

Reply:
Something that caught my attention was the `IOException`, since it seems to be out of the level of abstraction this class is on.
Since this class is not returning any object like an `OutputStream` or things like that, the `IOException` is out of place, being a more low level exception.

If this class uses an object that throws this exception, I think it should catch it and return a more meaningful exception based on the logic of this class.

And so on with all the other classes. We should make sure that the exceptions we declare on the methods are at the same level of abstraction as the class itself. Otherwise, let's create custom exceptions, catch the more low level ones and throw the specific ones.

What do you guys think? @wilmerrootstock , @marcos-iov , @nathanieliov , @julia-zack , @apancorb .

Reply:
Done! Guys, I removed them since I didn't find a place where they were being thrown.